### DISCUSSION TITLE
How do I split the source file into host and device to compile separately

### DISCUSSION THREAD

**Discussion by malixian** [intel/llvm#3244] [Q&A] [answered]
I found the using the option `-fsycl-device-only` can only compile the SYCL device code. I want to find a way to only compile the SYCL host code.

**Comment by keryell** [comment, marked as answer]
Since SYCL is pure C++, if you compile the SYCL code with a normal host compiler, you should have the host code and it will work also on the host CPU.
Is this what you are locking for?
This is called *host fallback* mode in the SYCL jargon.

**Reply by AlexeySachkov** [reply]
@keryell is right, in order to compile only host part of the SYCL code, just compile it without any `-fsycl` flags, i.e. like it is a regular C++ app. You can find some examples in [sycl/test/separate-compile](https://github.com/intel/llvm/tree/sycl/sycl/test/separate-compile)
